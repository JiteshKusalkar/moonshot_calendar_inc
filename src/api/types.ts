type LaunchRequestParams = {
  limit: number;
  offset: number;
  include_suborbital?: boolean;
  is_crewed?: boolean;
  related?: boolean;
  window_start__gt?: string;
  window_end__gt?: string;
};

type PadLocation = {
  id: number;
  country_code: string;
  name: string;
  url: string;
  total_landing_count: string;
  total_launch_count: number;
  map_image: string;
};

type Pad = {
  id: string;
  latitude: string;
  longitude: string;
  location: PadLocation;
};

type Launch = {
  pad: Pad;
};

type LaunchResponse = {
  results: Launch[];
  count: number;
  next: string;
  previous: string;
};

export type { Launch, LaunchResponse, LaunchRequestParams };
