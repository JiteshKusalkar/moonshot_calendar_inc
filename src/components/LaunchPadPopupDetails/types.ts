import { Launch } from '../../api/types';

type LaunchPadPopupDetailsProps = {
  launch?: Launch;
};

export type { LaunchPadPopupDetailsProps };
