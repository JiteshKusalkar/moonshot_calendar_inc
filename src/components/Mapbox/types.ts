import { LatLngTuple } from 'leaflet';
import { ReactNode } from 'react';

type LaunchPadMarker = {
  id: string;
  position: LatLngTuple;
  popupMessage: ReactNode;
};

type MapboxProps = {
  launchPadsLocation: LaunchPadMarker[];
};

export type { LaunchPadMarker, MapboxProps };
