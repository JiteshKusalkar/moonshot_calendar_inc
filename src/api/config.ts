function get(endpoint: string) {
  const url = `${process.env.REACT_APP_BASE_URL}/${process.env.REACT_APP_URL_VER}/${endpoint}`;

  return fetch({ method: 'GET', url } as RequestInfo);
}

export { get };
