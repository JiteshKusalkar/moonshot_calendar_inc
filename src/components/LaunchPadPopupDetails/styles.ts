import styled from "styled-components";

const Wrapper = styled.div``;
const ReadonlyField = styled.div`
  display: flex;
  align-self: center;
`;
const Label = styled.span`
  font-weight: 500;
  margin-right: 5px;
`;
const Value = styled.span``;

export { Label, ReadonlyField, Value, Wrapper };
