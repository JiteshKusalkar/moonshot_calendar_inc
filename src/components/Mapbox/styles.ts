import styled from 'styled-components';

const Wrapper = styled.div`
  .leaflet-container {
    height: 100vh;
  }
`;

export { Wrapper };
