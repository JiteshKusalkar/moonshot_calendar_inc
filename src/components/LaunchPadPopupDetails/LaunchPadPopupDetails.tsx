import { Label, ReadonlyField, Value, Wrapper } from './styles';
import { LaunchPadPopupDetailsProps } from './types';

function LaunchPadPopupDetails({ launch }: LaunchPadPopupDetailsProps) {
  return (
    <Wrapper>
      <ReadonlyField>
        <Label>Name:</Label>
        <Value>{launch?.pad.location.name || '-'}</Value>
      </ReadonlyField>
      <ReadonlyField>
        <Label>Name of pad:</Label>
        <Value>{launch?.pad.name || '-'}</Value>
      </ReadonlyField>
      <ReadonlyField>
        <Label>Agency:</Label>
        <Value>{launch?.launch_service_provider.name || '-'}</Value>
      </ReadonlyField>
    </Wrapper>
  );
}

export default LaunchPadPopupDetails;
