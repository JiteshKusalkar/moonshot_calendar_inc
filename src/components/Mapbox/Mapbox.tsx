import L, { LatLngTuple } from 'leaflet';
import { MapContainer, Marker, Popup, TileLayer } from 'react-leaflet';

import { Wrapper } from './styles';

import iconMarker from 'leaflet/dist/images/marker-icon.png';
import iconShadow from 'leaflet/dist/images/marker-shadow.png';
import { MapboxProps } from './types';

let DefaultIcon = L.icon({
  iconUrl: iconMarker,
  shadowUrl: iconShadow,
});

L.Marker.prototype.options.icon = DefaultIcon;

const token = process.env.REACT_APP_MAPBOX_ACCESS_TOKEN;
const styleId = process.env.REACT_APP_STYLE_ID
const userId = process.env.REACT_APP_USER_ID
const position: LatLngTuple = [51.505, -0.09];

// mapbox://styles/reet48/ckdaqo17l1gcs1iqgvf6rp9in

const url = `https://api.mapbox.com/styles/v1/${userId}/${styleId}/tiles/256/{z}/{x}/{y}@2x?access_token=${token}`;

function Mapbox(props: MapboxProps) {
  return (
    <Wrapper>
      <MapContainer center={position} zoom={2} scrollWheelZoom={false}>
        <TileLayer
          attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          url={url}
        />
        {props.launchPadsLocation.map(({ id, position, popupMessage }) => (
          <Marker key={id} position={position}>
            <Popup>{popupMessage}</Popup>
          </Marker>
        ))}
      </MapContainer>
    </Wrapper>
  );
}

export default Mapbox;
