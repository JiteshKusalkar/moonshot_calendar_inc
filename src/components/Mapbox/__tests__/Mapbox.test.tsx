import { render } from '@testing-library/react';
import 'jest-styled-components';

import { Launch } from '../../../api/types';
import LaunchPadPopupDetails from '../../LaunchPadPopupDetails';
import Mapbox from '../Mapbox';
import { MapboxProps } from '../types';

function MockFunction(props: any) {
  return <div>{props.children}</div>;
}

jest.mock('react-leaflet', () => ({
  MapContainer: MockFunction,
  Marker: MockFunction,
  Popup: MockFunction,
  TileLayer: MockFunction,
}));

describe('LaunchPadMap', () => {
  const launch = {
    launch_service_provider: {
      id: 115,
      name: 'Arianespace',
      type: 'Commercial',
    },
    pad: {
      id: 83,
      name: 'Cosmodrome Site 1S',
      latitude: '51.884395',
      longitude: '128.333932',
      location: {
        id: 18,
        url: 'https://lldev.thespacedevs.com/2.2.0/location/18/',
        name: 'Vostochny Cosmodrome, Siberia, Russian Federation',
        country_code: 'RUS',
        map_image:
          'https://spacelaunchnow-prod-east.nyc3.digitaloceanspaces.com/media/launch_images/location_18_20200803142401.jpg',
        total_launch_count: 11,
        total_landing_count: 0,
      },
    },
  };

  const props: MapboxProps<Launch> = {
    launchPadsLocation: [
      {
        id: 83,
        position: [51.884395, 128.333932],
        data: launch,
      }
    ],
    renderMessage: launchProp => <LaunchPadPopupDetails launch={launchProp} />
  }
  const { container } = render(<Mapbox {...props} />);

  it('should render the component', () => {
    expect(container).toMatchSnapshot();
  });
});
