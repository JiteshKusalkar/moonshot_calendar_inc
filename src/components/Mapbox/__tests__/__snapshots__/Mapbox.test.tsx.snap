// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchPadMap should render the component 1`] = `
<div>
  <div
    class="iZEJDz"
  >
    <div>
      <div />
      <div>
        <div>
          <div
            class=""
          >
            <div
              class="kCqMgU"
            >
              <span
                class="kTxztz"
              >
                Name:
              </span>
              <span
                class=""
              >
                Vostochny Cosmodrome, Siberia, Russian Federation
              </span>
            </div>
            <div
              class="kCqMgU"
            >
              <span
                class="kTxztz"
              >
                Name of pad:
              </span>
              <span
                class=""
              >
                Cosmodrome Site 1S
              </span>
            </div>
            <div
              class="kCqMgU"
            >
              <span
                class="kTxztz"
              >
                Agency:
              </span>
              <span
                class=""
              >
                Arianespace
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
